Fixed readme file (#44)
jv-feb24-group-project2/accommodation-booking-app: README.md

|---|---|-------|---|
|POST  | /api/auth/register | ---   |  Register a new user  |
|POST | /api/auth/login  | All   |  Authenticate a user  |
|  GET | /api/users/me  | USER   | Get user profile information  |
|  PUT  |/api/users/me   | USER  | Update user profile information  |
| PUT  |  /api/users/3/role | ADMIN | Update user roles |

- ##### Accommodations

|  HTTP method  |  Endpoint |  Role  | Description  |
|---|---|---|---|

> | GET  |  /api/accommodations | All  |  Get accommodations from catalog |

Reviewer: ```suggestion
| GET  |  /api/accommodations | All  |  Get page of accommodations |
```

> | GET  |  /api/accommodations/{id} | All | Get accommodation for id|

Reviewer: ```suggestion
| GET  |  /api/accommodations/{id} | All | Get accommodation by ID|
```

|  POST | /api/accommodations   |  MANAGER| Add a new accommodation |

> |  PUT  |/api/accommodations/{id}  |  MANAGER | Update a specific accommodation|
> | DELETE   |  /api/accommodations/{id}|   MANAGER |  Delete a specific accommodation |

Reviewer: ```suggestion
|  PUT  |/api/accommodations/{id}  |  MANAGER | Update accommodation by ID|
| DELETE   |  /api/accommodations/{id}|   MANAGER |  Delete accommodation by ID|
```

- ##### Amenity 

|  HTTP method  |  Endpoint |  Role | Description  |
|---|---|---|---|

> | GET  |  /api/amenities | All  |  Get all accommodations' amenities |
> | GET  | /api/amenities/{id}  | All   |  Get an amenity by id |

Reviewer: ```suggestion
| GET  |  /api/amenities | --- |  Get page of accommodations' amenities |
| GET  | /api/amenities/{id}  | --- |  Get an amenity by ID |
```

|  POST | /api/amenities  |  ADMIN | Add a new amenity  |

> |  PUT  |/api/amenities/{id}   |  ADMIN | Update a current category  |
> | DELETE   |  /api/amenities/{id} |  ADMIN|  Delete a current category |

Reviewer: ```suggestion
|  PUT  |/api/amenities/{id}   |  ADMIN | Update category by ID |
| DELETE   |  /api/amenities/{id} |  ADMIN|  Delete category by ID |
```

- ##### Payment 
| HTTP method | Endpoint  | Role | Description  |
|-------------|---|------|---|
| GET         |  /api/payments | USER |  Get payments history of certain customer |

> | GET         |   /api/payments/{id}| USER | Get the payment by id  |

Reviewer: ```suggestion
| GET         |   /api/payments/{id}| USER | Get the payment by ID  |
```

| GET         |  /api/payments/create-session/{bookingId} | ---  | Init payment sessions for booking transactions  |
| GET         | /api/payments/success/  | ---  |  Redirection after successfully processing of payment |

> | GET         |/api/payments/cancel/   | ---  | Redirection after unsuccessfully processing of payment  |

Reviewer: Not shore about this description. Better ask from Taras.

- ##### Booking
| HTTP method | Endpoint           |   Role  | Description                             |
|-------------|--------------------|---|-----------------------------------------|
| POST        | /api/bookings      | USER   | Creat new accommodation bookings        |

> | GET         | /api/bookings/my   | USER  | Get booking history of certain customer |

Reviewer: ```suggestion
| GET         | /api/bookings/my   | USER  | Get booking history of current customer |
```

> | GET         | /api/bookings/{id} |  MANAGER | Get a current booking                   |

Reviewer: ```suggestion
| GET         | /api/bookings/{id} |  MANAGER | Get booking by ID                  |
```

> | GET         |  /api/bookings            | MANAGER  |       Retrieves bookings based on user ID and their status                                   |

Reviewer: ```suggestion
| GET         |  /api/bookings            | MANAGER  |       Get page of bookings with filtration by user ID and/or booking status                                   |
```

> | PUT         |/api/bookings/{id}        | MANAGER  | Update a current booking                |
> | DELETE      |/api/bookings/{id}        | MANAGER  | Delete a current booking                |

Reviewer: ```suggestion
| PUT         |/api/bookings/{id}        | MANAGER  | Update booking by ID               |
| DELETE      |/api/bookings/{id}        | MANAGER  | Delete booking by ID               |
```

Solution: Understanding the intricacies of payment session creation proved crucial to ensuring a seamless payment process.

### 👥 Contributors:
- Artem Akymenko **[LinkedIn](https://www.linkedin.com/in/artem-akymenko-4b143030a/)** , **[GitHub](https://github.com/k0sm0naft)** 

> - Taras Fedak **[LinkedIn](https://www.linkedin.com/in/taras-fedak-a4535b147/?utm_source=share&utm_campaign=share_via&utm_content=profile&utm_medium=ios_app)** , **[GitHub](https://github.com/fedaktaras)** 

Reviewer: ```suggestion
- Taras Fedak **[LinkedIn](https://www.linkedin.com/in/taras-fedak-a4535b147/)** , **[GitHub](https://github.com/fedaktaras)** 
```
